openwisp_monitoring/check/tasks.py

```diff
@@ -54,3 +53,21 @@ def auto_create_ping(model, app_label, object_id, created):
     check = Check(name='Ping', check=ping_path, content_type=ct, object_id=object_id)
     check.full_clean()
     check.save()
+
+
+@shared_task
+def find_related_checks(object_id):
+    """
+    Retrieves all related checks to the passed ``device``
+    and calls the ``perform_check`` task from each of them.
+    If no check exists changes the status to ``OK``.
+    """
+    iterator = Check.objects.filter(active=True,
```

Comment:
to get the checks related to a device, we may use the reverse generic relation I mentioned above.